1. Before applying this commit:
[root@localhost pydbus-0.6.0]# make check
make: *** No rule to make target 'check'.  Stop.
[root@localhost pydbus-0.6.0]#
[root@localhost pydbus-0.6.0]# sh tests/run.sh python3
Main: t1
Lol: t2
Main: t1
Lol: t2
M1
M2
Obj: 1
Obj: 2
asyn: Obj: 1
Obj: 3
asyn: Obj: 2
sync: Obj: 3
Obj: 4
asyn: Obj: 4
[root@localhost pydbus-0.6.0]#

    
2. After applying this commit:
[root@localhost pydbus-0.6.0]# make check
python3 setup.py install
running install
......
Processing dependencies for pydbus==0.6.0
Finished processing dependencies for pydbus==0.6.0
sh tests/run.sh python3
=====================================================================
TEST START
Obj: 1
Obj: 2
Obj: 3
asyn: Obj: 1
sync: Obj: 3
asyn: Obj: 2
Obj: 4
asyn: Obj: 4
PASS: tests/publish_async.py
PASS: tests/publish_error.py
M1
M2
PASS: tests/publish_multiface.py
PASS: tests/publish_properties.py
Lol: t2
Main: t1
Lol: t2
Main: t1
PASS: tests/publish.py
TEST END
=====================================================================
TEST RESULT SUMMARY for python-pydbus
\# TOTAL: 5
\# PASS:  5
\# FAIL:  0
=====================================================================

--- a/Makefile
+++ b/Makefile
@@ -1,5 +1,7 @@
 packages: wheel sdist
 
+PYTHON=python
+
 wheel:
 	rm -Rf build
 	./setup.py bdist_wheel
@@ -7,3 +9,7 @@ wheel:
 sdist:
 	rm -Rf build
 	./setup.py sdist
+
+check:
+	${PYTHON} setup.py install
+	sh tests/run.sh ${PYTHON}

--- a/tests/run.sh
+++ b/tests/run.sh
@@ -13,9 +13,28 @@ PYTHON=${1:-python}
 "$PYTHON" $TESTS_DIR/error.py
 if [ "$2" != "dontpublish" ]
 then
-	"$PYTHON" $TESTS_DIR/publish.py
-	"$PYTHON" $TESTS_DIR/publish_properties.py
-	"$PYTHON" $TESTS_DIR/publish_multiface.py
-	"$PYTHON" $TESTS_DIR/publish_async.py
-	"$PYTHON" $TESTS_DIR/publish_error.py
+	echo "====================================================================="
+	echo "TEST START"
+	all=0
+	xfail=0
+	xpass=0
+	for test_file in `ls $TESTS_DIR/publish*.py`
+	do
+	    all=`expr $all + 1`
+	    "$PYTHON" $test_file
+	    if [ $? -eq 0 ]; then
+	       echo "PASS: $test_file"
+	       xpass=`expr $xpass + 1`
+	    else
+	       echo "FAIL: $test_file"
+	       xfail=`expr $xfail + 1`
+	    fi
+	done
+	echo "TEST END"
+	echo "====================================================================="
+	echo "TEST RESULT SUMMARY for python-pydbus"
+	echo "# TOTAL: $all"
+	echo "# PASS:  $xpass"
+	echo "# FAIL:  $xfail"
+	echo "====================================================================="
 fi

